src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/IRemoteDebuggerAuthenticationService.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
+{
+    [ProjectSystemContract(ProjectSystemContractScope.UnconfiguredProject, ProjectSystemContractProvider.Private)]
+    internal interface IRemoteDebuggerAuthenticationService
+    {
+        IEnumerable<IRemoteAuthenticationProvider> GetRemoteAuthenticationModes();
+        IRemoteAuthenticationProvider? GetProviderForAuthenticationMode(string remoteAuthenticationMode);
+        bool ShowRemoteDiscoveryDialog(ref string remoteDebugMachine, ref IRemoteAuthenticationProvider? remoteAuthenticationProvider);
```

Comment:
Should should be marked as not returning null if return is true?

Reply:
No reason the dialog couldn't return something we don't know about.

Reply:
So in that case we return the remote machine, but throw away the authentication provider? What situation would we want that?

Reply:
I would want that in all situations - doing the opposite is quite literally throwing out the baby with the bathwater.

Think about a user who uses this dialog, types in their machine name and picks the "Foo" auth type, which is unsupported. If we throw away both pieces of data, they press OK and nothing happens. With the current code, the machine name changes, and the auth mode changes to a blank entry to indicate that its not in the drop down. Inspecting the drop down also clearly indicates that "Foo" isn't listed.

Neither of those is a perfect scenario, I'll grant you, but one leads to a "the dialog does nothing" bug report, and one leads to "Foo is not in the drop down", the latter being much easier to deal with. We could show a message box if the auth provider isn't found, so they'd know we were throwing away their data, but a message box like that should be shown by the dialog, not by us, as once we've shown it its too late for them to recover, they just have to go back into the dialog and try again, which is also a bad experience. This is where the "UnknownAuthProvider" would have come in in the past, but I removed that as its not something that we can provide when the auth mode guids and port supplier guids are different.